### PR TITLE
Profiles: adding support to capture prefered language

### PR DESCRIPTION
**Pull request by AkhileshNegi**
## Summary
 Target issue is #2401  
 
## Checklist
  Before submitting a pull request, please ensure that you mark these task.

- [x] Ran `mix setup` in the repository root and test.
- [x] If you've fixed a bug or added code that is tested and has test cases. 
- [ ] In the case of adding a new API, you added a **postman** request for that.
- [x] Coding standard and conventions are followed. https://docs.google.com/document/d/1rfU33IjS-ioiIH0TBWpxoLsdyyYdI1tDrqr5HCRIb98/edit?usp=sharing

## Notes

- Added support to capture user preferred language at profile level when language is switched
- Added Testcases

### CODE REVIEW

File: lib/glific/flows/contact_setting.ex [hunk 1]
```diff
@@ -41,13 +42,35 @@ defmodule Glific.Flows.ContactSetting do
...
+  @spec maybe_update_profile_language(Contact.t(), non_neg_integer()) ::
+          Contact.t()
+  def maybe_update_profile_language(
+        %{active_profile_id: active_profile_id} = contact,
+        language_id
+      )
+      when is_integer(active_profile_id) do
+    with profile <- Profiles.get_profile!(active_profile_id) do
+      Profiles.update_profile(profile, %{language_id: language_id})
+    end
```
Comment (pankaj-ag) [inline lib/glific/flows/contact_setting.ex]: Looks like you are in love with the **with**  statement. 😊

I could not find any difference between the below code
 ```elixir
 Profiles.get_profile!(active_profile_id)
 |> Profiles.update_profile(%{language_id: language_id})
 ```
and your code.

**Review by pankaj-ag — CHANGES_REQUESTED** [review]

File: lib/glific/flows/contact_setting.ex (hunk 1 above)
Comment (AkhileshNegi) [reply lib/glific/flows/contact_setting.ex]: @pankaj-ag restructured it a bit 😇

**Review by pankaj-ag — APPROVED** [review]
@AkhileshNegi Approving this. 
but it's always a good idea to use the centralised method instead of using fetch_by. 

That will helps us in case we want to implement some ACL (Permissions) on a single resource.

**Comment by AkhileshNegi** [comment]
@pankaj-ag Good point. I prefer fetch_by as it returns ok tuple when resource is found and returns error tuple when not found. which helps in making sure the resource is there before moving down the flow. Let me know if there is a better way. Happy to explore😇

**Comment by pankaj-ag** [comment]
> 

yes, that's why we throw an error in the case of get_resource functions. Because if we have an id and don't have a record for that in the DB it means there is some inconsistency which we should handle by cascading operations and callbacks on the databases or correcting the request itself. 

We can use fetch_by to get the results for label, shortcode and things like that since those are user inputs and the system doesn't set that by itself.

**Pull request merged**